api/app/routes/kie.py

```diff
@@ -3,27 +3,47 @@
 # This program is licensed under the Apache License 2.0.
 # See LICENSE or go to <https://opensource.org/licenses/Apache-2.0> for full license details.
 
-from typing import Dict, List
+from typing import List
 
-from fastapi import APIRouter, File, UploadFile, status
+from fastapi import APIRouter, File, HTTPException, UploadFile, status
 
-from app.schemas import OCROut
+from app.schemas import KIEElement, KIEOut
 from app.vision import kie_predictor
-from doctr.io import decode_img_as_tensor
+from doctr.io import DocumentFile
 
 router = APIRouter()
 
 
-@router.post("/", response_model=Dict[str, List[OCROut]], status_code=status.HTTP_200_OK, summary="Perform KIE")
-async def perform_kie(file: UploadFile = File(...)):
+@router.post("/", response_model=List[KIEOut], status_code=status.HTTP_200_OK, summary="Perform KIE")
+async def perform_kie(files: List[UploadFile] = [File(...)]):
     """Runs docTR KIE model to analyze the input image"""
-    img = decode_img_as_tensor(file.file.read())
-    out = kie_predictor([img])
-
-    return {
-        class_name: [
-            OCROut(box=(*prediction.geometry[0], *prediction.geometry[1]), value=prediction.value)
-            for prediction in out.pages[0].predictions[class_name]
-        ]
-        for class_name in out.pages[0].predictions.keys()
-    }
+    results: List[KIEOut] = []
+    for file in files:
+        mime_type = file.content_type
+        if mime_type in ["image/jpeg", "image/png"]:
+            content = DocumentFile.from_images([await file.read()])
+        elif mime_type == "application/pdf":
+            content = DocumentFile.from_pdf(await file.read())
+        else:
+            raise HTTPException(status_code=400, detail=f"Unsupported file format for KIE endpoint: {mime_type}")
```

Comment:
You can reduce duplicated code here by creating a function which read `file.content_type` and load the doc